### PR TITLE
Add exchange bitrue

### PR DESCRIPTION
**Pull request by HartS**
This PR builds on top of the work in PR #7943 by @firepol (and previous work by @dust and @binares) to add Bitrue support to ccxt. The [contribution guidelines](https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-contribute-code) instruct us to avoid contributing changes from the build; however, I did not add these files (or remove them) on my branch; I can remove them or revert commits which add them if requested.

Notes about the changes made:

- Implicit methods are now all being `await`ed (many of the methods previously didn't work due to this)
- `ArgumentsRequired` is thrown when trying to call a method without the symbol argument when the exchange requires it (fetchOrder and cancelOrder for example)
- `fetchClosedOrders` has been removed; Based on [this comment](https://github.com/ccxt/ccxt/issues/6881#issuecomment-623117208) from @kroitor , it's better not to emulate it when it's not offered by the exchange
- fetchBalance now calls parseBalance with the `legacy` argument set to true due to more recent changes in ccxt (I didn't investigate this too much, but it didn't work without it)
- Call `parseOrder()` on `privateDeleteOrder()` response in `cancelOrder()`

See the bulk of the work in 4621037da0557d67949b31de8a7b401b90d737c1 for some more details on my changes.

I tested the following methods to ensure everything is working

- [x] `fetchTicker`
- [x] `fetchTickers`
- [x] `fetchOrderBook`
- [x] `fetchTrades` (bitrue API does not include associated order info from this endpoint)
- [x] `fetchBalance`
- [x] `createOrder` Note: The bitrue API does not include some order info, such as limit and side, in the response here. Rather than add it (since we already have this from the creation params), I left it empty. I can add this info from the params to `createOrder(...)` if requested
- [x] `cancelOrder`
- [x] `fetchOrder`
- [x] `fetchOpenOrders`
- [x] `fetchOrders`

Please let me know if there's anything else you'd like me to assist with to get this merged.

### CODE REVIEW

**Comment by HartS** [comment]
Added a small update to hopefully fix CI issues. One thing I noticed while looking at how parseTickers works is that a new market is not being parsed/recognized: `LIQUSDT`, which results in an `undefined` key in the result from `fetchTickers`. Can this be fixed by adding `LIQ` to the `commonCurrencies` object?

**Comment by HartS** [comment]
Tests passing after last commit to fix signatures of fetchOrders and fetchOpenOrders

**Review by DUK3135 — APPROVED** [review]

File: python/ccxt/async_support/bitrue.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            market = self.market(symbol)
+        request = {}
+        if symbol is not None:
+            request['symbol'] = market['id']
+        if start_id is not None:
+            request['fromId'] = start_id
+        if limit is not None:
+            request['limit'] = limit
+        response = self.privateGetMyTrades(self.extend(request, params))
+        trades = response if isinstance(response, (list)) else []
```
Comment (mrbaguvix) [inline python/ccxt/async_support/bitrue.py:341]: Should always be a list right?
```suggestion
        trades = response if isinstance(response, list) else []
```

**Review by mrbaguvix — CHANGES_REQUESTED** [review]

File: python/ccxt/async_support/bitrue.py (hunk 1 above)
Comment (HartS) [reply python/ccxt/async_support/bitrue.py:341]: Hi @mrbaguvix , I believe this was generated. I also didn't commit this file (I mentioned in the PR description that the compiled assets were added by previous contributors, whose work I started from), so perhaps @firepol can weigh in here. The python files will have to be rebuilt when the PR is ready to be merged (or post-merge, I'm not sure how that works).

The contribution guidelines state:

```
There is now a single source version in one language, that is JavaScript. Other language-specific versions are syntactically derived (transpiled, generated) automatically from the source version. 
```

So I took this to mean that I shouldn't commit any changes from rebuilding.

Comment (HartS) [reply python/ccxt/async_support/bitrue.py:341]: in fact, the section I linked in the description instructs us *not* to commit python and other transpiled assets.

**Review by mrbaguvix — APPROVED** [review]
Good to go I guess @kroitor ?

**Comment by tammycamp** [comment]
Hey @kroiter, would be great to get a timeline on this pull request.  We would love to use it.

@HartS, thank you for your contribution.

**Pull request merged**